bot/models/class_models.py

```diff
@@ -0,0 +1,59 @@
+from dataclasses import dataclass
+from datetime import datetime
+
+from bot.utils.helpers import strtodt
+
+
+@dataclass
+class ClassSemester:
+    semester_id: str
+    semester_name: str
+    semester_start: str
+    semester_end: str
+
+    def start_date(self) -> datetime:
+        return strtodt(self.semester_start)
+
+    def end_date(self) -> datetime:
+        return strtodt(self.semester_end)
+
+
+@dataclass
+class ClassChannelScaffold:
+    class_prefix: str
+    class_number: int
+    class_professor: str
+    class_name: str
+
+    def class_code(self) -> str:
+        return f'{self.class_prefix}-{self.class_number}'
+
+    def full_title(self) -> str:
+        return f'{self.class_code()}: {self.class_name}'
+
+    def channel_name(self) -> str:
+        return f'{self.class_code().lower()}-{self.class_professor.lower()}'
+
+    def intended_category(self) -> str:
+        level = (self.class_number // 1000) * 1000
+        return f'{self.class_prefix} {level} LEVELS'
```

Comment:
Don't do this, use f-string formatting 
<img width="421" alt="image" src="https://user-images.githubusercontent.com/38477514/221326039-b3426481-9d5f-4a67-8a04-6d2ce44c3986.png">

Reply:
This has the downside of making it basically unreadable, plus it's taking an already large number (ex. `2120`, `4910`) and getting the thousandths place and adding three 0's basically. Not sure how you would do that with f-string formatting.

Reply:
ah shiii i misread 1 sec